frontend/javascripts/admin/job/job_list_view.js

```diff
@@ -0,0 +1,162 @@
+  activeUser: ?APIUser,
+|};
+type Props = { ...StateProps, history: RouterHistory };
+
+type State = {
+  isLoading: boolean,
+  jobs: Array<APIJob>,
+  searchQuery: string,
+};
+
+const persistence: Persistence<State> = new Persistence(
+  { searchQuery: PropTypes.string },
+  "jobList",
+);
+
+class JobListView extends React.PureComponent<Props, State> {
+  intervalID;
+
+  state = {
+    isLoading: true,
+    jobs: [],
+    searchQuery: "",
+  };
+
+  componentWillMount() {
+    this.setState(persistence.load(this.props.history));
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  componentWillUpdate(nextProps, nextState) {
+    persistence.persist(this.props.history, nextState);
+  }
+
+  componentWillUnmount() {
+    clearTimeout(this.intervalID);
+  }
+
+  async fetchData(): Promise<void> {
+    const jobs = await getJobs();
+    this.setState(
+      {
+        isLoading: false,
+        jobs: jobs || [],
+      },
+      // refresh jobs every 5 seconds
+      () => {
+        this.intervalID = setTimeout(this.fetchData.bind(this), 5000);
+      },
+    );
+  }
+
+  handleSearch = (event: SyntheticInputEvent<>): void => {
+    this.setState({ searchQuery: event.target.value });
+  };
+
+  render() {
+    return (
+      <div className="container">
+        <div style={{ marginTag: 20 }}>
+          <div className="pull-right">
+            <Search
+              style={{ width: 200 }}
+              onPressEnter={this.handleSearch}
+              onChange={this.handleSearch}
+              value={this.state.searchQuery}
+            />
+          </div>
+          <h3>Jobs</h3>
+          <div className="clearfix" style={{ margin: "20px 0px" }} />
+
+          <Spin spinning={this.state.isLoading} size="large">
+            <Table
+              dataSource={Utils.filterWithSearchQueryAND(
+                this.state.jobs,
+                ["datasetName"],
+                this.state.searchQuery,
+              )}
+              rowKey="id"
+              pagination={{
+                defaultPageSize: 50,
+              }}
+              style={{ marginTop: 30, marginBotton: 30 }}
+            >
+              <Column
+                title="Job Id"
+                dataIndex="id"
+                key="id"
+                sorter={Utils.localeCompareBy(typeHint, job => job.id)}
+              />
+              <Column
+                title="Description"
+                key="datasetName"
+                render={job => `${job.type} of ${job.datasetName}`}
+              />
+              <Column
+                title="Created at"
+                key="createdAt"
+                render={job => new Date(job.createdAt).toUTCString()}
+                sorter={Utils.compareBy(typeHint, job => job.createdAt)}
+              />
+              <Column
+                title="State"
+                dataIndex="state"
+                key="state"
+                sorter={Utils.localeCompareBy(typeHint, job => job.state)}
```

Comment:
Could you format the `state` from "SUCCESS" to "Success" (etc.)? The capslock formatting looks a bit rough :)

Reply:
Is all lower case okay, too?

Reply:
I think, capitalized would be nicer. You can use `_.capitalize` for that :)

Reply:
That's neat, thanks!